msteams-platform/m365-apps/actions-in-m365.md

```diff
@@ -0,0 +1,213 @@
+
+#### Update app manifest
+
+Define the intent, object, and handler for your actions in the app manifest schema (previously called Teams app manifest).
+
+The following is an app manifest example for with `intent` and `supportedobjects` and `handlers` properties for `openPage` and `openDialog`:
+
+* Action to open a page: Action shows related tasks in the **To-do** app based on the selected file. It uses `"intent": "custom"` to identify the file type, such as .xlsx or doc. The `"type": "openPage"` handler opens the app and navigates to the `pageId` and `subpageId` (need more information in code line no 96).
+
+    ```json
+    "actions": [
+        {
+            // Defining Action to open a page
+            "id": "relatedTasks",
+            "displayName": "Related tasks",
+            "intent": "custom",
+            "description": "Shows tasks in the To do app that are related to this file.",
+            "handlers": [
+                {
+                    "type": "openPage",
+                    "supportedObjects": {
+                        "file": {
+                            "extensions": ["xlsx", "doc", "docx", "pdf", "pptx", "ppt"]
+                        }
+                    },
+                    "pageInfo": {
+                        "pageId": "index",
+                    }
+                }
+            ]
+        }
+    ]
+    ```
+
+* Action to open a dialog: Action lets you add a selected file and a note to the **To-do** app. It uses an `"intent": "addTo"` to identify the file type, such as .docx or .doc. The `"type": "openDialog"` handler opens a web-based dialog from the specified URL.
+
+    ```json
+    "actions": [
+        {
+            // Defining Action to open a dialog
+            "id": "addTodoTask",
+            "displayName": "Add todo task",
+            "intent": "addTo",
+            "description": "Add this file with a short note to my to do list",
+            "handlers": [
+                {
+                    "type": "openDialog",
+                    "supportedObjects": {
+                        "file": {
+                            "extensions": ["xlsx", "doc", "dot", "docx", "pdf", "pptx", "ppt"]
+                        }
+                    },
+                    "dialogInfo": {
+                        "dialogType": "url",
+                        "url": "http://localhost:53000/index.html#/dialogPage",
+                        "width": "small",
+                        "height": "large"
+                    }
+                }
+            ]
+        }
+    ]
+    ```
+
+For more information, see [public developer preview app manifest schema](../resources/schema/manifest-schema-dev-preview.md#actions).
+
+#### Create a handler
+
+Update the handler to receive the Action information through the context object to create a seamless user experience for performing users specific tasks using the [Teams JavaScript library (TeamsJS)](/javascript/api/@microsoft/teams-js).
```

Comment:
The section heading says "Create a handler" but the text here implies a handler already exists. This section should provide guidance and code samples on building a handler--this info is included in Step 2 of the private preview doc.